File: php/tests/test_10min.php
<?php
declare(strict_types=1);

require_once __DIR__ . '/../src/Exceptions/TempMailException.php';
require_once __DIR__ . '/../src/Exceptions/RateLimitException.php';
require_once __DIR__ . '/../src/Exceptions/NotFoundException.php';
require_once __DIR__ . '/../src/Models/Message.php';
require_once __DIR__ . '/../src/Models/MessageDetail.php';
require_once __DIR__ . '/../src/HttpClient.php';
require_once __DIR__ . '/../src/TempMailProvider.php';
require_once __DIR__ . '/../src/TempMailFactory.php';

foreach (glob(__DIR__ . '/../src/Providers/*.php') as $file) {
    require_once $file;
}

use TempMail\TempMailFactory;

$provider = TempMailFactory::create('10minutemail');
echo "Generating email...\n";
$email = $provider->generateEmail();
echo "Generated email: $email\n";

echo "Fetching inbox...\n";
$inbox = $provider->getInbox($email);
echo "Inbox count: " . count($inbox) . "\n";
foreach ($inbox as $msg) {
    echo "Message ID: {$msg->id}, Sender: {$msg->sender}, Subject: {$msg->subject}\n";
    echo "Reading message...\n";
    $detail = $provider->readMessage($email, $msg->id);
    echo "Read message subject: {$detail->subject}\n";
    echo "Read message bodyText: " . substr($detail->bodyText, 0, 100) . "...\n";
}
echo "Done!\n";

File: java/src/test/java/com/tempmail/Test10Min.java
package com.tempmail;

import com.tempmail.providers.TenMinuteMailProvider;
import com.tempmail.models.Message;
import com.tempmail.models.MessageDetail;
import java.util.List;

public class Test10Min {
    public static void main(String[] args) throws Exception {
        TenMinuteMailProvider provider = new TenMinuteMailProvider();
        System.out.println("Generating email...");
        String email = provider.generateEmail();
        System.out.println("Generated email: " + email);

        System.out.println("Fetching inbox...");
        List<Message> inbox = provider.getInbox(email);
        System.out.println("Inbox size: " + inbox.size());
        for (Message m : inbox) {
            System.out.println("Message ID: " + m.getId() + ", Sender: " + m.getSender() + ", Subject: " + m.getSubject());
            System.out.println("Reading message...");
            MessageDetail detail = provider.readMessage(m.getId());
            System.out.println("Read message subject: " + detail.getSubject());
            System.out.println("Read message bodyText: " + detail.getBodyText());
        }
        System.out.println("Done!");
    }
}
